Basic_Python_Diploma/YaDisk.py

```diff
@@ -193,47 +193,70 @@ def _check_existance(param):
 
     def download(self, item):
         """Метод скачивает на жесткий диск файл или папку"""
-        self._point()
-        target_folder = 'downloads'
-        if not os.path.exists(target_folder):
-            os.mkdir(target_folder)
-        try:
-            if item.type == 'dir':
-                target_folder = os.path.join(target_folder, item.path.split('disk:/')[-1])
-                if not os.path.exists(target_folder):
-                    os.mkdir(target_folder)
-                target_folder = os.path.abspath(target_folder)
-            else:
-                target_folder = os.path.abspath(target_folder)
-                file_to_download = requests.get(item.link)
-                with open(os.path.join(target_folder, item.name), 'wb') as file:
-                    file.write(file_to_download.content)
-        except (KeyError, AttributeError):
-            if item['type'] == 'dir':
-                target_folder = os.path.join(target_folder, item['path'].split('disk:/')[-1])
-                if not os.path.exists(target_folder):
-                    os.mkdir(target_folder)
-                target_folder = os.path.abspath(target_folder)
         try:
-            param = {'path': item.path}
-        except (KeyError, AttributeError):
-            param = {'path': item['path']}
+            item_name = item.path.split('disk:/')[-1]
+            item_type = item.type
+            item_path = item.path
+            item_title = item.name
+        except AttributeError:
+            item_name = item['path'].split('disk:/')[-1]
+            item_type = item['type']
+            item_path = item['path']
+            item_title = item['name']
+
+        if item_type == "dir":
+            print(item_type)
+            target_folder = os.path.abspath(os.path.join('downloads', item_name))
+            print(target_folder)
+            try:
+                os.makedirs(target_folder, exist_ok=True)
+            except FileExistsError:
+                return "Файл уже существует на диске"
+        else:
+            print(item_type)
+            target_folder = os.path.abspath(os.path.join('downloads', item_name.split(item_name)[0]))
```

Comment:
Непустая строка, разделенная по себе самой, всегда будет возвращать список из двух пустых строк:

```
In [6]: item_name = "qwe"                                                                                                                                                                                                                             

In [7]: item_name.split(item_name)                                                                                                                                                                                                                            
Out[7]: ['', '']
```

Поэтому эту строку можно упростить вот так:

```suggestion
            target_folder = os.path.abspath(os.path.join('downloads', ''))
```

А это в свою очередь можно упростить ещё сильнее вот так:

```suggestion
            target_folder = os.path.abspath('downloads')
```